lib/game.rb

```diff
@@ -0,0 +1,128 @@
+
+class Game
+
+    attr_accessor  :board, :player_1, :player_2
+
+
+    WIN_COMBINATIONS =  [
+        [0,1,2],
+        [3,4,5],
+        [6,7,8],
+        [0,3,6],
+        [1,4,7],
+        [2,5,8],
+        [0,4,8],
+        [6,4,2]
+    ]
+
+    def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+         @player_1 = player_1
+         @player_2 = player_2
+         @board = board
+    end
+
+    def current_player
+       @board.turn_count % 2 == 0 ? @player_1 : @player_2
+    end
+
+    def over?
+        if won? != false
+            return true
+        end
+        if draw?
+            return true
+        end
+         return false
+    end
+
+    def won?
+        win = WIN_COMBINATIONS.find do |combo|
+             @board.cells.values_at(combo[0],combo[1],combo[2]).all?{|v| v == "X"}  || @board.cells.values_at(combo[0],combo[1],combo[2]).all?{|v| v == "O"}
+         end
+          if win == nil
+              return false
+          elsif win != nil
+              win
+          end
+    end
+
+    def draw?
+        @board.full? && won? == false ? true : false
+    end
+
+    def winner
+        won? == false ? nil : @board.cells[won?[0]]
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.